File: Assets/Samples/ModuTree/JankenSample/Nodes/IsPlayerChoseCondition.cs
using System.Threading;
using System.Threading.Tasks;
using ModuTree.Runtime.Core;
using ModuTree.Runtime.Nodes;

namespace ModuTree.JankenSample
{
    /// <summary>
    /// プレイヤーが指定の手を選んでいるかチェックするコンディション。
    /// targetHand と Blackboard の playerHand が一致すれば Success を返す。
    /// </summary>
    [BehaviourNodeMeta(
        displayName: "プレイヤーの手チェック",
        category:    "Condition",
        description: "プレイヤーが targetHand を選んでいれば Success",
        color:       "#2E5E2E")]
    public class IsPlayerChoseCondition : ConditionNodeData
    {
        [NodeField("対象の手", "この手を選んでいれば Success")]
        public JankenHand targetHand = JankenHand.Rock;

        protected override Task<NodeState> OnUpdateAsync(CancellationToken ct)
        {
            var playerHand = Blackboard.Get(JankenBBKeys.PlayerHand);
            var result = playerHand == targetHand ? NodeState.Success : NodeState.Failure;
            return Task.FromResult(result);
        }
    }
}

File: Assets/Samples/ModuTree/JankenSample/PHP/Nodes/IsPlayerChoseCondition.php
<?php
declare(strict_types=1);

/**
 * プレイヤーが指定の手を選んでいるかチェックするコンディション。
 * targetHand と Blackboard の playerHand が一致すれば Success を返す。
 *
 * JSONパラメータ例:
 *   {"targetHand": 0}  // JankenHand::Rock
 *   {"targetHand": 1}  // JankenHand::Scissors
 *   {"targetHand": 2}  // JankenHand::Paper
 */
class IsPlayerChoseCondition extends ConditionNodeData
{
    /** 比較対象の手。JSONから自動復元される（int→JankenHand変換）。 */
    public JankenHand $targetHand = JankenHand::Rock;

    protected function onUpdate(): NodeState
    {
        /** @var JankenHand|null $playerHand */
        $playerHand = $this->blackboard->get(JankenBBKeys::$playerHand);
        return $playerHand === $this->targetHand
            ? NodeState::Success
            : NodeState::Failure;
    }
}